janusgraph-core/src/main/java/org/janusgraph/core/ConfiguredGraphFactory.java

```diff
@@ -155,10 +155,16 @@ public static JanusGraph close(String graphName) throws Exception {
      * @throws BackendException If an error occurs during deletion
      * @throws ConfigurationManagementGraphNotEnabledException If ConfigurationManagementGraph not
      */
-    public static void drop(String graphName) throws Exception {
-        final StandardJanusGraph graph = (StandardJanusGraph) ConfiguredGraphFactory.close(graphName);
+    public static void drop(String graphName) throws BackendException, ConfigurationManagementGraphNotEnabledException, Exception {
+        final ConfigurationManagementGraph configManagementGraph = getConfigGraphManagementInstance();
+        configManagementGraph.removeConfiguration(graphName);
+
+        final JanusGraphManager jgm = JanusGraphManagerUtility.getInstance();
+        Preconditions.checkNotNull(jgm, JANUS_GRAPH_MANAGER_EXPECTED_STATE_MSG);
+
+        final StandardJanusGraph graph = (StandardJanusGraph) jgm.getGraph(graphName);
+        removeGraphFromCache(graph);
```

Comment:
> The bug was in the call to removeGraphFromCache(String graphName) which internally tries to open the graph. Because of the order of the calls, the graph was already dropped by a call to JanusGraphFactory.drop(graph) which then result in the storage to be recreated.

This sentence confuses me a bit, due the fact you didn't change the functionality of it.

Can you clarify it for me?

Reply:
There are two versions of `removeGraphFromCache()` one expects a [String graphName](https://github.com/JanusGraph/janusgraph/blob/89c18d053b81a45145a834e81e53f847abfa30d4/janusgraph-core/src/main/java/org/janusgraph/core/ConfiguredGraphFactory.java#L231) and the other expects a [JausGraph graph](https://github.com/JanusGraph/janusgraph/blob/89c18d053b81a45145a834e81e53f847abfa30d4/janusgraph-core/src/main/java/org/janusgraph/core/ConfiguredGraphFactory.java#L243).

The previous version of this method used the version that expects a _String graphName_ which [open the graph](https://github.com/JanusGraph/janusgraph/blob/89c18d053b81a45145a834e81e53f847abfa30d4/janusgraph-core/src/main/java/org/janusgraph/core/ConfiguredGraphFactory.java#L234) and call the other version that expect a `JanusGraph` instance.

The problem with that approach is that the `open()` call will create the graph if it doesn't exist, with the previous implementation the graph was dropped before calling `removeGraphFromCache()` which caused the graph to be created again.

What I did was use the version of `removeGraphFromCache()` that expect a `JanusGraph` instance which doesn't need to open the graph.

Reply:
Thanks for explanation